elastic/kibana
# [Unified Integrations] Create Services, Storybook, Replacements Card; add to Fleet

## Summary

This PR adds a new component, as well as the infrastructure for more shared components moving forward:

- Creates a Storybook setup for `customIntegrations`;
- Uses the Presentation Util [Plugin Services Toolkit](https://docs.elastic.dev/kibana-dev-docs/presentationPlugin#plugin-services-toolkit) for creating shared services;
- Adds a new service for Replacement Integrations;
- Creates a new component for Replacement Integrations;
- Adds Storybook stories;
- Integrates component into `fleet` detail pages.

## Testing
1. Run `yarn storybook custom_integrations` to see this component on its own.
  > You can also click "Storybooks Preview" -> "customIntegrations" to see the static build from CI.
2. Run `yarn storybook fleet` to see this component on the Details page in Fleet.
  > You can also click "Storybooks Preview" -> "fleet" to see the static build from CI.
3. Run Kibana and visit an Integration to see it against live data.

## Component in Storybook
<img width="755" alt="Screen Shot 2021-10-05 at 12 04 02 AM" src="https://user-images.githubusercontent.com/297604/135963439-39ca1669-58a5-4d52-9a57-2c5d46a55fd7.png">

## Fleet Addition in Storybook
<img width="1449" alt="Screen Shot 2021-10-05 at 12 07 18 AM" src="https://user-images.githubusercontent.com/297604/135963530-34918d0a-5be1-4e8c-b6d5-bdb0922f1317.png">

## Fleet Addition in Kibana
<img width="1655" alt="Screen Shot 2021-10-05 at 12 06 04 AM" src="https://user-images.githubusercontent.com/297604/135963550-c3c6cf45-778b-4eed-ade8-b9208d7338f5.png">
